src/osrm/osrm.cpp

```diff
@@ -15,7 +16,18 @@ namespace osrm
 
 // Pimpl idiom
 
-OSRM::OSRM(engine::EngineConfig &config) : engine_(std::make_unique<engine::Engine>(config)) {}
+OSRM::OSRM(engine::EngineConfig &config)
+{
+    if (engine::Engine<engine::algorithm::CoreCH>::CheckCompability(config))
```

Comment:
I was planning on adding a flag `use_mld` for when we generalize this for MLD. I don't think it makes sense to use vanilla CH data when you could be using CoreCH: It just makes things a lot slower by not using the extra data that is there.